### PR TITLE
Added contact filter by role and get providers that are not contacts …

### PR DESCRIPTION
**Pull request by ppazos**
Added features for story https://www.pivotaltracker.com/story/show/109084232

The providerController index will be used to fill the "add to contacts" screen on the apps, listing the providers that are not contacts of the patient.

### CODE REVIEW

File: grails-app/controllers/io/tarmac/api/HealthcareProvidersController.groovy [hunk 1]
```diff
@@ -12,12 +14,60 @@ import static org.springframework.http.HttpStatus.NOT_FOUND
 class HealthcareProvidersController extends ApiController{
     def contactService
 
-    def index() {
-        try{
+    /**
+     * if notInContacts it returns only the providers that are not contacts of the authenticated patient.
+     */
+    def index(boolean notInContacts)
```
Comment (ascuccimarra) [inline grails-app/controllers/io/tarmac/api/HealthcareProvidersController.groovy]: Insted of having an endpoint specially for healthcare providers (that may or may not be contacts) we could just use the /contacts?role.. for getting an user contacts, and have a /potential_contacts?role... for getting users that are not contacts. Also ContactService:: findPotentialContacts takes care of that I think.

Comment (ppazos) [inline grails-app/controllers/io/tarmac/api/HealthcareProvidersController.groovy]: @ascuccimarra I thought about that. Getting providers that are not contacts form the contacts endpoint didn't make much sense. Let me check the findPotentialContacts.

**Comment by ppazos** [comment]
@ascuccimarra I have added the notInContacts filter to the /contacts endpoint
I keeped the coded under /providers?notInContacts because iit does less processing than the findPotentialDoctors and returns Person instead of ContactData, but results are consistent on both.

**Comment by ascuccimarra** [comment]
+1

**Comment by ppazos** [comment]
I'll checkout master and reapply my changes it is still has conflicts.

**Pull request closed without merging**